VultisigApp/VultisigApp/View Models/AppViewModel.swift

```diff
@@ -30,14 +30,37 @@ class AppViewModel: ObservableObject {
     
     private let logic = AccountLogic()
     
+    static let shared = AppViewModel()
+    
     func restart() {
-        set(selectedVault: nil)
+        set(selectedVault: selectedVault, restartNavigation: true)
     }
     
-    func set(selectedVault: Vault?, showingVaultSelector: Bool = false) {
-        self.selectedVault = selectedVault ?? self.selectedVault
+    func set(selectedVault: Vault?, showingVaultSelector: Bool = false, restartNavigation: Bool = false) {
```

Comment:
`restartNavigation` you should set default to `true` because previously it will reset the Navigation stack by default